apps/andi/lib/andi/input_schemas/datasets/extract_steps/extract_date_step.ex

```diff
@@ -71,18 +73,34 @@ defmodule Andi.InputSchemas.Datasets.ExtractDateStep do
     end
   end
 
-  defp validate_format(changeset) do
+  defp validate_timex_format(changeset) do
     put_change(changeset, :format, "{YYYY}-{0M}-{0D} {h24}:{m}:{s}")
   end
 
   defp validate_time_unit(%{changes: %{deltaTimeUnit: unit}} = changeset)
        when unit in ["microseconds", "milliseconds", "seconds", "minutes", "hours", "days", "weeks", "months", "years", ""],
        do: changeset
 
-  defp validate_time_unit(%{changes: %{deltaTimeUnit: unit}} = changeset), do: add_error(changeset, :deltaTimeUnit, "invalid time unit")
+  defp validate_time_unit(%{changes: %{deltaTimeUnit: _unit}} = changeset), do: add_error(changeset, :deltaTimeUnit, "invalid time unit")
 
   defp validate_time_unit(changeset), do: changeset
 
+  defp validate_delta_change(%{changes: %{deltaTimeUnit: deltaTimeUnit}} = changeset) when deltaTimeUnit not in [nil, ""] do
+    case get_change(changeset, :deltaTimeValue) in [nil, ""] do
+      true -> add_error(changeset, :deltaTimeValue, "must be set when deltaTimeUnit is set")
+      false -> changeset
+    end
+  end
+
+  defp validate_delta_change(%{changes: %{deltaTimeValue: deltaTimeValue}} = changeset) when deltaTimeValue not in [nil, ""] do
```

Comment:
what's the difference between this and the above function?

Reply:
the one above checks for `deltaTimeUnit` in the changeset and verifies that `deltaTimeValue` is also set. We are basically validating that if one is set, then the other has to be too, even though neither of them are required

Reply:
🤦‍♂️